github_api_helper.js

```diff
@@ -27,24 +28,41 @@ function inProgressAPIresponse (context, headSha) {
 }
 
 /**
- * Filters the listed files and returns only the files relevant to us
- * @param {*} rawData the raw output from list files API call
- * @return {Promise[]<any>} the filtered GitHub response
+ * Get Precaution config file contents as raw data,
+ * if that file exists.
+ * @param {import('probot').Context} context Probot context
+ * @return {Promise<any>} GitHub response
+ * See https://developer.github.com/v3/repos/contents/#get-contents
  */
-function filterData (rawData) {
-  return rawData.data.filter(file => config.fileExtensions.reduce((acc, ext) => acc || file.filename.endsWith(ext), false))
-    .filter(fileJSON => fileJSON.status !== 'removed').map(fileInfo => {
-      return fileInfo.filename
+async function getConfigFile (context) {
+  let { owner, repo } = context.repo()
+  // GitHub doesn't provide "checkIfFileExists" API endpoint.
+  // That's why we should try to download the file and if we have
+  // 404 error code the file doesn't exist in the repository.
+  try {
+    return await context.github.repos.getContents({
+      owner,
+      repo,
+      path: config.configFilePath,
+      headers: { accept: rawMediaType }
     })
+  } catch (error) {
+    if (error.code === 404) {
+      return undefined
```

Comment:
it would be cleaner if we always returned the same type from this function, let's return an empty array `[]` rather than `undefined`. Is there some JS convention I'm unaware of that makes returning `undefined` better here?

Reply:
First, it's important to notice that because this function is async it will always return a promise:

_The word “async” before a function means one simple thing: a function always returns a promise. Even If a function actually returns a non-promise value, prepending the function definition with the “async” keyword directs JavaScript to automatically wrap that value in a resolved promise._
from https://javascript.info/async-await

So, even if I return '[]' in that situation we would return basically Promise<[]> instead of Promise<undefine> where in <> is the value of the promise. 
I used "undefined" here because in JavaScript when a variable doesn't have any value is marked as "undefined" and because if there is no config file in the repo then confObj makes sense to be undefined.

Another bonus is that it's easy to check if a value has undefined value like here in index.js
```
// Loads the Precaution configuration file
const rawObj = await apiHelper.getConfigFile(context, cache.getBranchPath(repoID, PR.id))
const configObj = rawObj ? JSON.parse(rawObj.data) : []
```

Reply:
Ah, OK. This is me not being JS savvy then. I had assumed `[] ? 'foo' : 'bar'` would give 'bar' much as `undefined ? 'foo' : 'bar'` does, much as in Python an empty array evaluates to `False`, but reading your comment and checking in a JS console I realise I assumed incorrectly.